### DISCUSSION TITLE
Yunohost was the help I needed to self-host

### DISCUSSION THREAD

**Discussion by P-Godfroid** [outline/outline#5072] [Self Hosting]
Hello,

I tried for severals days to make it work and I failed each time.

We have a ubuntu server, we have a dns domain and everything needed (except we don't use slack, but creating a free account was easy).

Each time a different problem, either minio version, or outline package restarting infinitely or even an "easy install" (https://chsasank.com/outline-self-hosted-wiki.html) which cannot produce a functional nginx configuration.

I have personnally tried something like 6 or 8 different tutorials, and crossover between them without any successful attempt.

I have several others projects still running, so I can't waste more time being stuck like that so we are ready to pay someone to install or rather 'help install by directly supervising the operation" a self-hosting https functional outline wiki.

I am looking forward to hearing from you,

Pierre

**Comment by Limezy** [comment]
Hi, have you tried [the Yunohost app](https://install-app.yunohost.org/?app=outline) ? (disclaimer, I'm the maintainer of that package)
The problem I see with paying someone is that you'll have troubles each time you want to upgrade your instance.
Yunohost is providing a safer environnement where everything is frozen to certain versions. You may have to wait longer to get the package updated, but you'll have good support within the community !

**Reply by P-Godfroid** [reply]
Oh, I didn't properly read the Yunohost documentation. It could work. Let me try.

About the problem you have mentioned, I am pretty sure I can replicate and maintain the server myself. Modify the versions of the software in a .yml is far from being difficult. Creating a functional one is the real challenge in this case.

**Reply by P-Godfroid** [reply]
@Limezy Ok, it seems like it works, but I am now stuck in a loop on the authentification procedure.
![image](https://user-images.githubusercontent.com/109596710/226431225-9c42ea13-1848-4746-984d-deb60d89c8c3.png)
The connect as a member button return to this :
![image](https://user-images.githubusercontent.com/109596710/226431744-d9722766-b55d-44dc-bbb7-0fd368843dd5.png)
and clicking on outline on this new page returns to the previous page.

**Reply by Limezy** [reply]
Can you please try to void all cookies and local storage ? I believe you have some remaining from your previous installation attempts in your browser cache, and this causes problems.

**Reply by P-Godfroid** [reply]
The same happen in a private window in a different browser.

**Reply by Limezy** [reply]
Did you secure all your domains with a SSL certificate already ?

**Reply by P-Godfroid** [reply]
Absolutely, we have a domain so all four domains already have the right DNS (1 for yunohost, 3 for outline) and I used "yunohost domain cert install"

**Reply by Limezy** [reply]
Ok, sorry to hear that. Let me try to do a fresh install on my test server and see what things are blocking. 
Meanwhile could you please check the logs for the dex service or the outline service in the admin panel ?

**Reply by P-Godfroid** [reply]
Bullpoint : 
There was a dex installation error in the Outline installation log:

![image](https://user-images.githubusercontent.com/109596710/226626953-31ac02a7-a34c-4b94-9cf8-bea58150c4bf.png)

Here is the dex installation log link:

https://paste.yunohost.org/raw/otihedunit

**Reply by Limezy** [reply]
... You have been victim of a rare event where the Dex installation is aborted due to a network delay... 
Two choice : 
1. You start again from scratch, after removing Outline (easier)
2. You install the Dex app using 
* The domain you previously set as domain
* Outline as oidc name
* The output from the command `sudo yunohost app setting outline oidc_secret` as the secret
* The output from the command `sudo yunohost app setting outline oidc_callback` as the callback

**Reply by P-Godfroid** [reply]
Hello,

I tried both possibilities.
When reinstalling Ouline, the same dex installation error occurred.

When installing only dex, I could not target the outline domain for dex as outline was using it.
![Capture d'écran 2023-03-22 092349](https://user-images.githubusercontent.com/109596710/226842513-8165ab5c-063f-4d3d-88b4-d88d73e38dd5.png)
For information I let the default /dex as URI.

I tried targeting the main yunohost domain instead and got this error :
https://paste.yunohost.org/raw/osoximopad

**Reply by Limezy** [reply]
For the dex installation error, that's very strange, let me look into it.
Please be reminded that you need three different root domains. 
Dex **can't** be installed on the same domain as Outline.

**Reply by Limezy** [reply]
I truly hope you're gonna be able to install Outline, this package has been working flawlessly since nearly a year now !

**Reply by P-Godfroid** [reply]
So I have one domain for outline, one domain for minio, one for yunohost so can I assume the authentification related one needed in the outline application is supposed to be the dex one  ?

I tried with it and still got the same error :

![image](https://user-images.githubusercontent.com/109596710/226902191-434fa09f-4846-4311-970b-8934729103cb.png)

**Reply by Limezy** [reply]
You need three domains : one for Outline, one for MinIO, one for Dex.
The Outline domain can be also set as your Yunohost main domain (this is how I do it on my server). I have : 
- `mydomain.fr`, on which Outline is installed on the root, and set as Yunohost main domain
- `auth.mydomain.fr` on which Dex is installed (it could be `auth.mydomain.fr/dex` but that's complicated for nothing)
- `docs.mydomain.fr` on which MinIO is installed

For the error, can you please share your `/tmp/go-build.20230322075040.120007.log` ?
I have just installed successfully Dex on my test server. Is your connectivity good ?

**Reply by P-Godfroid** [reply]
So I used the right domain so far.

As far as I know, the connectivity is top notch. In the yunohost diagnostic, only the blocked port 25 and the IPV6 reverse proxy return errors and I don't think these can create the issue.

![image](https://user-images.githubusercontent.com/109596710/226921933-33e0c608-7aa1-4060-b993-682d6e4f18a9.png)
I am pretty sure it won't help, there is nothing new in this log.

**Reply by P-Godfroid** [reply]
@Limezy 

I successfully installed it. I did two things differently than prior:

- Connect with a user instead of the main administrator before starting the installation.

- Target the main yunohost page as main page for the outline app.

However, I still have a question.

It seems like the yunohost outline is a different version as I can't use the Json  format for importing.
Is there an updated version expected ?


Anyway, a big thanks for the help.

**Reply by P-Godfroid** [reply]
Btw, a little French translation error here :

![image](https://user-images.githubusercontent.com/109596710/228172505-6d79cf83-4f52-4ad1-83ff-875dcc421da9.png)

It would be : 

"Processus lancé" instead" of "Vous demandé"

**Reply by Limezy** [reply]
Hi, I'm very glad that you managed to install Outline.

- I'm still quite unsure about what happened, feel free to open an issue on https://github.com/Yunohost-Apps/outline_ynh if you have a minute (for my future me... who knows when!).
- Let me correct the translation mistake on Crowdin
- I have my own branch regularly keeping my own server up to date with the main branch of Outline here, however I'm ususally waiting for the official community release before I push that. What I can do is push my current version to the testing branch of that repo, and you could try to migrate (at your own risk, please backup properly before !) using `sudo yunohost app upgrade outline --debug -u https://github.com/yunohost-Apps/outline_ynh/tree/testing`. Watch for changes on that branch in the coming hours

**Reply by P-Godfroid** [reply]
Ok, I will do a server backup and will try it.
Some settings like the theme preference are nice to have.

Thanks again

**Reply by Limezy** [reply]
CI is running : https://ci-apps-dev.yunohost.org/ci/job/6436
Will merge the current version to testing branch once done
https://github.com/YunoHost-Apps/outline_ynh/pull/72
This will ship 68640860fb9cf584f155348ae06e4536226bef16

**Reply by Limezy** [reply]
The Yunohost package is currently 98% depending on my time, which is far from being extensible. I'll be happy if you are willing to help with it. It's usually pretty straightforward to bump from one version to another.
Once I'll get some more time I'll also have to migrate to Yunohost package v2 format, but as of now it's not yet mandatory

**Reply by Limezy** [reply]
... and don't forget to sponsor Outline, even if symbolic

**Reply by P-Godfroid** [reply]
> ... and don't forget to sponsor Outline, even if symbolic

I will definitely do it. It is a nice software.

About the help, it will depend of the language and the required knowledge if I can directly help, but I will see what I can do, contact me in mp about that.

**Comment by tommoor** [comment]
A note that if you wish to purchase the [enterprise license](https://www.getoutline.com/pricing) then installation support is included.

**Reply by P-Godfroid** [reply]
Hello,

1) Does it includes the possibility to select our own domain ?
For the 10$ cloud version for example we tried with a trial we need to upgrade to 30$ to be able to use our own domain.
Without this constraint, we would have gone for the 10$/month version.

2) Does user means wiki visitor or only contributors ?

3) Does user means concurrent wiki contributor ?

4) What about the minimum purchase of 100 seats ?

**Reply by tommoor** [reply]
**Does it includes the possibility to select our own domain ?**
Yes.

**Does user means wiki visitor or only contributors ?**
Anyone who signs in with authentication.

**What about the minimum purchase of 100 seats ?**
The minimum purchase size covers the cost of providing installation support.
